From: Alex Gonzalez <alexg@balena.io>
Date: Mon, 24 Oct 2022 18:05:47 +0200
Subject: [PATCH 7/9] rtc-unipi: Adapt to removal of nvram ABI

Adapt to the removal of the nvram ABI in commit
25ece30561d2 (rtc: nvmem: remove nvram ABI)

Upstream-status: pending
Signed-off-by: Alex Gonzalez <alexg@balena.io>
---
 modules/rtc-unipi/rtc-unipi.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/modules/rtc-unipi/rtc-unipi.c b/modules/rtc-unipi/rtc-unipi.c
index 255bcd181824..6615b8fa7491 100644
--- a/modules/rtc-unipi/rtc-unipi.c
+++ b/modules/rtc-unipi/rtc-unipi.c
@@ -671,11 +671,13 @@ read_rtc:
 
 		rtc_unipi->rtc->nvmem_config = &rtc_unipi->nvmem_cfg;
 		rtc_unipi->rtc->nvram_old_abi = true;
-#else
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,17,0) && LINUX_VERSION_CODE < KERNEL_VERSION(5,10,0)
 		nvmem_cfg.priv = rtc_unipi;
 	    rtc_unipi->rtc->nvram_old_abi = true;
 	    rtc_nvmem_register(rtc_unipi->rtc, &nvmem_cfg);
-
+#else
+	    nvmem_cfg.priv = rtc_unipi;
+	    rtc_nvmem_register(rtc_unipi->rtc, &nvmem_cfg);
 #endif
 	rtc_unipi->rtc->ops = &mcp794xx_rtc_ops; /*chip->rtc_ops ?: &ds13xx_rtc_ops;*/
 	err = rtc_register_device(rtc_unipi->rtc);
